src/main/java/com/fasterxml/jackson/databind/deser/std/JsonNodeDeserializer.java

```diff
@@ -765,7 +765,17 @@ protected final JsonNode _fromFloat(JsonParser p, DeserializationContext ctxt,
         if (nt == JsonParser.NumberTypeFP.BIG_DECIMAL) {
             return _fromBigDecimal(ctxt, nodeFactory, p.getDecimalValue());
         }
-        if (ctxt.isEnabled(DeserializationFeature.USE_BIG_DECIMAL_FOR_FLOATS)) {
+
+        // [databind#4801] Add JsonNodeFeature.USE_BIG_DECIMAL_FOR_FLOATS
+        DatatypeFeatures dtf = ctxt.getDatatypeFeatures();
+        boolean useBigDecimal;
+        if (dtf.isExplicitlySet(JsonNodeFeature.USE_BIG_DECIMAL_FOR_FLOATS)) {
```

Comment:
This is not what was agreed in the issue. I do not have my computer right now so can't add the links. I thought it was agreed that this only had an effect is set to true (enabled). If set to false or null then the DeserializationFeature value is used.

Reply:
Ah I thought we agreed to [later comment](https://github.com/FasterXML/jackson-databind/issues/4801#issuecomment-2746669263). Explicitly setting `JsonNodeFeature.USE_BIG_DECIMAL_FOR_FLOATS` would take precedence over more global `DeserializationFeature.USE_BIG_DECIMAL_FOR_FLOATS` one.

Reply:
@pjfanning No. If it is explicitly set, it should be used as-is, being more specific (only affects `JsonNode`s) than general-purpose `DeserializationFeature`.
This one change I pointed out based on your original proposal.

Without this, it would not be possible to use `BigDecimal` for all other types (or mostly just `Number` or `Object`) but `Double` for `JsonNode`s.

Reply:
* ok - so we have https://github.com/FasterXML/jackson-databind/issues/4801#issuecomment-2766597667 with the change that matches what this code does
* I guess it's fine to go this way